Add a few new preserve-3d hit-testing tests for currently broken behaviour.

mattwoodrow opened the pull request:
#### 3eb1748df76d1fa53f1b170dc4ec54e653edd870
<pre>
Add a few new preserve-3d hit-testing tests for currently broken behaviour.
<a href="https://bugs.webkit.org/show_bug.cgi?id=248281">https://bugs.webkit.org/show_bug.cgi?id=248281</a>
&lt;rdar://problem/102632565&gt;

Reviewed by Simon Fraser.

Neither of these tests should change behaviour with interop, but we can fix them at the same
time.

* LayoutTests/platform/gtk/transforms/3d/point-mapping/3d-point-mapping-2-expected.txt:
* LayoutTests/platform/ios/transforms/3d/point-mapping/3d-point-mapping-2-expected.txt:
* LayoutTests/platform/mac/transforms/3d/point-mapping/3d-point-mapping-2-expected.txt:
* LayoutTests/transforms/3d/point-mapping/3d-point-mapping-2.html:

This adds a new subtest check for hitting the parent box3 element in the area where it&apos;s
drawn on top of the child. We&apos;re currently failing this, as we hit-test box4 despite it
being rendered behind.

* LayoutTests/transforms/3d/preserve-3d-not-applied-to-ancestors-expected.txt: Added.
* LayoutTests/transforms/3d/preserve-3d-not-applied-to-ancestors.html: Added.

This test has a transformed ancestor of the preserve-3d element, which shouldn&apos;t be
participating in transform accumulation (and rendering already shows that it isn&apos;t).
The bounding client rect should match, and it currently doesn&apos;t.

Canonical link: <a href="https://commits.webkit.org/257091@main">https://commits.webkit.org/257091@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/aaf71050e89088054e7421e7ecda6499418bd38e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/97545 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/6800 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/30712 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/107048 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/167312 "Built successfully and passed tests") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/7141 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/35558 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/89944 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/103716 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/103198 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/5349 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/84178 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/32354 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/87224 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/89014 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/75278 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/800 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/20473 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/788 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/21951 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4869 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/5603 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/44422 "Found 1 new test failure: http/wpt/service-workers/fetch-service-worker-preload.https.html (failure)") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/2039 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/41329 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

smfr reviewed (approved):
Should these be WPT?

mattwoodrow committed:
Add a few new preserve-3d hit-testing tests for currently broken behaviour.
https://bugs.webkit.org/show_bug.cgi?id=248281
<rdar://problem/102632565>

Reviewed by Simon Fraser.

Neither of these tests should change behaviour with interop, but we can fix them at the same
time.

* LayoutTests/platform/gtk/transforms/3d/point-mapping/3d-point-mapping-2-expected.txt:
* LayoutTests/platform/ios/transforms/3d/point-mapping/3d-point-mapping-2-expected.txt:
* LayoutTests/platform/mac/transforms/3d/point-mapping/3d-point-mapping-2-expected.txt:
* LayoutTests/transforms/3d/point-mapping/3d-point-mapping-2.html:

This adds a new subtest check for hitting the parent box3 element in the area where it's
drawn on top of the child. We're currently failing this, as we hit-test box4 despite it
being rendered behind.

* LayoutTests/transforms/3d/preserve-3d-not-applied-to-ancestors-expected.txt: Added.
* LayoutTests/transforms/3d/preserve-3d-not-applied-to-ancestors.html: Added.

This test has a transformed ancestor of the preserve-3d element, which shouldn't be
participating in transform accumulation (and rendering already shows that it isn't).
The bounding client rect should match, and it currently doesn't.

Canonical link: https://commits.webkit.org/257091@main
